actool/manifest.go

```diff
@@ -0,0 +1,380 @@
+package main
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/appc/spec/aci"
+	"github.com/appc/spec/schema"
+	"github.com/appc/spec/schema/types"
+)
+
+var (
+	inputFile  string
+	outputFile string
+
+	patchNocompress   bool
+	patchOverwrite    bool
+	patchReplace      bool
+	patchManifestFile string
+	patchName         string
+	patchUser         string
+	patchGroup        string
+	patchCaps         string
+	patchMounts       string
+
+	catPrettyPrint bool
+
+	cmdPatchManifest = &Command{
+		Name:        "patch-manifest",
+		Description: `Copy an ACI and patch its manifest.`,
+		Summary:     "Copy an ACI and patch its manifest (experimental)",
+		Usage: `
+		  [--manifest=MANIFEST_FILE]
+		  [--name=example.com/app]
+		  [--user=uid] [--group=gid]
+		  [--capability=CAP_SYS_ADMIN,CAP_NET_ADMIN]
+		  [--mounts=work,path=/opt,readOnly=true[:work2,...]]
+		  [--replace]
+		  INPUT_ACI_FILE
+		  [OUTPUT_ACI_FILE]`,
+		Run: runPatchManifest,
+	}
+	cmdCatManifest = &Command{
+		Name:        "cat-manifest",
+		Description: `Print the manifest from an ACI.`,
+		Summary:     "Print the manifest from an ACI",
+		Usage:       ` [--pretty-print] ACI_FILE`,
+		Run:         runCatManifest,
+	}
+)
+
+func init() {
+	cmdPatchManifest.Flags.BoolVar(&patchOverwrite, "overwrite", false, "Overwrite target file if it already exists")
+	cmdPatchManifest.Flags.BoolVar(&patchNocompress, "no-compression", false, "Do not gzip-compress the produced ACI")
+	cmdPatchManifest.Flags.BoolVar(&patchReplace, "replace", false, "Replace the input file")
+
+	cmdPatchManifest.Flags.StringVar(&patchManifestFile, "manifest", "", "Manifest file")
```

Comment:
a bit ambiguous, prefer something like `Replace image manifest with this file. Incompatible with other replace options.`